Added carousel role description

agliga opened the pull request:
## Description
Added role description. Will generate an ID used for `aria-labelledby` on the root. This ID is only available when there is a clipped text (`a11yStatusText or `a11yHeadingText`, or is `discrete`).
Added `role="group"` and `aria-roledescription="carousel"`
Also appended title of carousel to buttons. 

## References
#609
#1177

ianmcburnie reviewed (changes requested):
Good news: the `aria-roledescription` change negates the need for appending the carousel title to each paddle, as the screen reader now automatically announces the roledescription (carousel) and label (top products).

Bad news: you'll need to revert the changes to paddle as it results in carousel and top products being announced twice  :-)

Also, we should remove any mention of the word carousel from our a11y text examples, as this is now handled by the screen reader via the roledescription value.

agliga commented:
> Good news: the `aria-roledescription` change negates the need for appending the carousel title to each paddle, as the screen reader now automatically announces the roledescription (carousel) and label (top products) when focus first moves inside the carousel group.
> 
> Bad news: you'll need to revert the changes to paddle as it results in carousel and top products being announced twice :-)
> 
> Also, we should remove any mention of the word carousel from our a11y text examples, as this is now handled by the screen reader via the roledescription value.

Done, reverted the change. Also I removed ` - carousel` from the default text

ianmcburnie commented:
Updates look good, but for some reason the first few examples don't have the ARIA attributes:

![Screen Shot 2020-09-03 at 9 18 31 PM](https://user-images.githubusercontent.com/38065/92199117-19ad1780-ee2b-11ea-86ca-21fb937348ea.png)

agliga commented:
> Updates look good, but for some reason the first few examples don't have the ARIA attributes:
> 
> ![Screen Shot 2020-09-03 at 9 18 31 PM](https://user-images.githubusercontent.com/38065/92199117-19ad1780-ee2b-11ea-86ca-21fb937348ea.png)
This is because I added a condition to not add `role=group` and `aria-describeby` if there is not `a11yText` or header text. Otherwise theres nothing to point `aria-labelledby` to and I thought that those two attributes dont work without aria-labelledby

ianmcburnie commented:
I see what you mean. I think role="group" and aria-roledescription="Carousel" can be ever present, but as you say for continuous case, the aria-labelledby will only be added when dev adds `a11y-heading-text` and `a11y-heading-tag`.

Oops, and yes I should be enjoying the day off, so please ignore this until next week ;-)

agliga commented:
> I see what you mean. I think role="group" and aria-roledescription="Carousel" can be ever present, but as you say for continuous case, the aria-labelledby will only be added when dev adds `a11y-heading-text` and `a11y-heading-tag`.
> 
> Oops, and yes I should be enjoying the day off, so please ignore this until next week ;-)

Done, and yes you should check this next week :+1:
